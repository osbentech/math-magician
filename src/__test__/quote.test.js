const { render, screen } = require('@testing-library/react');
require('@testing-library/jest-dom');
const Quote = require('../quote').default;

describe('Tests for Quote component', () => {
  it('Render Quote page', () => {
    const quote = render(<Quote />);
    expect(quote).toMatchSnapshot();
  });

  it("Test if the quote by Benjamin Osegbo exists", () => {
    render(<Quote />);
    expect(
      screen.getByText(
        /extraodinary is where brain meets magic/i,
      ),
    ).toBeInTheDocument();
  });
});
