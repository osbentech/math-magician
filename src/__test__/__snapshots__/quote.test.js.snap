// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for Quote component Render Quote page 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <header>
          <h1>
            A Benjamin think...
          </h1>
          <ul>
            <a
              href="/"
            >
              Home
            </a>
            <span>
               | 
            </span>
            <a
              href="/Calculator"
            >
              Calculator
            </a>
            <span>
               | 
            </span>
            <a
              href="/Quote"
            >
              Quotes
            </a>
          </ul>
        </header>
        <main>
          <h2>
            Hey|| BEN here|| officially welcoming you to my page
          </h2>
          <ul
            class="qNames"
          >
            <li>
              extraodinary is where brain meets magic (Benjamin Osegbo)
            </li>
            <li>
              It is better to fail in originality than to succeed in imitation.
            </li>
            <li>
              The road to success and the road to failure are almost exactly the same
            </li>
          </ul>
        </main>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <header>
        <h1>
          A Benjamin think...
        </h1>
        <ul>
          <a
            href="/"
          >
            Home
          </a>
          <span>
             | 
          </span>
          <a
            href="/Calculator"
          >
            Calculator
          </a>
          <span>
             | 
          </span>
          <a
            href="/Quote"
          >
            Quotes
          </a>
        </ul>
      </header>
      <main>
        <h2>
          Hey|| BEN here|| officially welcoming you to my page
        </h2>
        <ul
          class="qNames"
        >
          <li>
            extraodinary is where brain meets magic (Benjamin Osegbo)
          </li>
          <li>
            It is better to fail in originality than to succeed in imitation.
          </li>
          <li>
            The road to success and the road to failure are almost exactly the same
          </li>
        </ul>
      </main>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
